### ISSUE TITLE
Building on previous version of FFmpeg

### ISSUE THREAD

**Issue by NullVSNone** [m-ab-s/media-autobuild_suite#1764]
Switching to a separate issue as a generic bug - can close if not meant to be supported.
Continuing with @1480c1's suggestions.

[logs.zip](https://0x0.st/ivhP.zip)

This is with removing `glslang` and the `grep` line - it still seems as though the patch is failing? Does this patch not work when freezing ffmpeg?

```
FFmpeg-devel-lavf-srt-fix-build-fail-when-used-the-libsrt-1.4.1.patch
        Patch could not be applied with `git am`. Continuing without patching.
├ Changing options to comply to lgplv3...
├ Compiling static FFmpeg...
├ Running configure...
├ Running make...
Likely error (tail of the failed operation logfile):
src/libavformat/libsrt.c:337:50: note: each undeclared identifier is reported only once for each function it appears in
src/libavformat/libsrt.c: In function 'libsrt_setup':
src/libavformat/libsrt.c:401:5: warning: 'srt_socket' is deprecated [-Wdeprecated-declarations]
  401 |     fd = srt_socket(cur_ai->ai_family, cur_ai->ai_socktype, 0);
      |     ^~
In file included from src/libavformat/libsrt.c:24:
C:/media-autobuild_suite-master/local64/include/srt/srt.h:730:41: note: declared here
  730 | SRT_ATR_DEPRECATED_PX SRT_API SRTSOCKET srt_socket(int, int, int) SRT_ATR_DEPRECATED;
      |                                         ^~~~~~~~~~
make: *** [/build/ffmpeg-git/ffbuild/common.mak:60: libavformat/libsrt.o] Error 1
make failed. Check C:/media-autobuild_suite-master/build/ffmpeg-git/build-static-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
  Creating diagnostics file...
```

----------

What's done so far:

Remove `do_patch "https://gist.githubusercontent.com/1480c1/18f251a03b7657241c98cc8baf93a223/raw/0001-glslang-add-MachineIndependent.patch" am`
Remove `grep -q SRTO_STRICTENC libavformat/libsrt.c &&`

FFmpeg version: 1.4.1

**Comment by 1480c1** [comment]
you might want to just disable srt in the ffmpeg_options.txt or something since it probably would not be worth trying to make it work with the latest srt and ffmpeg

**Comment by NullVSNone** [comment]
Trying to keep SRT in there, so until https://trac.ffmpeg.org/ticket/8760 or https://github.com/Haivision/srt/issues/1382 are fixed, it's not a viable option build like this?

**Comment by 1480c1** [comment]
problem is, it's considered sort of fixed using the patchwork patch, but since the format of libsrt.c is different between the FFmpeg versions, it might not work as intended

you could try copying the libsrt.c from the latest FFmpeg master and overwrite then do the patch to just compile the latest libsrt.c

**Comment by NullVSNone** [comment]
> you could try copying the libsrt.c from the latest FFmpeg master and overwrite then do the patch to just compile the latest libsrt.c

Not sure how to do that, so I got [libsrt.c](https://git.ffmpeg.org/gitweb/ffmpeg.git/blob/b6663adaae948a66574dff58923a862774663439:/libavformat/libsrt.c) from the latest FFmpeg master. What do I overwrite? If I put it in ffmpeg-git won't that just get overwritten?

**Comment by 1480c1** [comment]
overwrite the same file in libavformat between do_vcs and the do_patch

**Comment by NullVSNone** [comment]
I didn't know what the line of code would look like. Also didn't know where it goes (between which do_vcs/do_patch).

Instead, I cloned the previous version of FFmpeg and used the latest libsrt.c - and that compiled properly. A lot of steps but that seemed to do it. Thank you for your help!

I think this story can be closed then?
